### PR TITLE
[Dataset] Fix CoLA dataset index and add Glue dataset API 

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
Added Glue datasets to the dataset API. Will refactor scripts/bert/dataset.py in a followup PR.

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/gluonnlp/data/glue.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+        super(GlueCoLA, self).__init__(root, data_file,
+            num_discard_samples=num_discard_samples, field_indices=field_indices)
+
+    def _repo_dir(self):
+        return 'gluon/dataset/GLUE/CoLA'
+
+@register(segment=['train', 'dev', 'test'])
+class GlueSST2(_GlueDataset):
+    """CoNLL2000 Part-of-speech (POS) tagging and chunking joint task dataset.
```
Comment (kenjewu) [inline src/gluonnlp/data/glue.py]: The document looks wrong

**Comment by mli** [comment]
Job PR-682/13 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-682/13/index.html

**Comment by eric-haibin-lin** [comment]
@Gpwner could you help review the CoLADataset class?

File: src/gluonnlp/data/glue.py [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+from ..base import get_home_dir
+
+_glue_s3_uri = 's3://apache-mxnet/gluon/dataset/Glue/'
+
+class _GlueDataset(TSVDataset):
+    def __init__(self, root, data_file, **kwargs):
+        root = os.path.expanduser(root)
+        if not os.path.isdir(root):
+            os.makedirs(root)
+        self._root = root
```
Comment (szha) [inline src/gluonnlp/data/glue.py]: nit: user may specify the root to be a path with `~` in it, so it might be a good idea to do `os.path.expanduser` here.

Comment (eric-haibin-lin) [reply src/gluonnlp/data/glue.py]: Anything else needed besides `os.path.expanduser(root)` at line 39?

Comment (szha) [reply src/gluonnlp/data/glue.py]: false alarm as I missed line 39

**Pull request merged**